### DISCUSSION TITLE
Error using "filter by column value" AND alias

### DISCUSSION THREAD

**Discussion by danilort** [dbeaver/dbeaver#12074] [Ideas]
#### System information: 
- Operating system (distribution) and version: Windows 10 versione 1909 build 18363.1316
- DBeaver version: DBeaver Community 7.3.5.202102151746 (and 7.3.0)
- Additional extensions: NULL

#### Connection specification:
- Database name and version: MySQL 8.0.21
- Driver name: Driver for MySQL 8 and later
- Do you use tunnels or proxies (SSH, SOCKS, etc): NULL

#### Steps to reproduce, if exist:

Steps to reproduce:
1) EXEC: SELECT SUBSTRING(richiesta, 1, 1) AS campo1 FROM accessi;
2) CLIC BUTTON: filter by column value
3) CLIC BUTTON: apply filter criteria
4) Errore SQL [1054] [42S22]: Unknown column 'campo1' in 'where clause'

if I replace the #2 with:
2) CLIC BUTTON: sort by column 
(OK, works well)

if I write the #1, without using alias:
1) EXEX: SELECT SUBSTRING(richiesta, 1, 1) FROM accessi;
(OK, works well)

**Comment by danilort** [comment]
Could you replace WHERE with HAVING?

**Comment by LonwoLonwo** [comment]
Hello @danilort 
We use column name for WHERE clause that the database with another metainformation returns to us. And this is 'campo1' in this case, unfortunately.
Also according [MySQL documentation](https://dev.mysql.com/doc/refman/8.0/en/select.html):

> Do not use HAVING for items that should be in the WHERE clause.

Although I understand that HAVING really works in this case.

The computation of WHERE value occurs earlier than the SELECT, in contrast to the sorting, which occurs at the end.

I'm afraid that this task, unfortunately do not have optimal solution. 
Perhaps you have other suggestions?

**Comment by danilort** [comment]
Hello Lon.
No, I have no other ideas.
If you want, you can close the ticket.
Thank you for your work.

Danilo

**Comment by LonwoLonwo** [comment]
I apologize that I could not help in this situation.
